turbinia/workers/containerd.py

```diff
@@ -0,0 +1,156 @@
+from turbinia.workers import TurbiniaTask
+
+log = logging.getLogger('turbinia')
+
+CE_BINARY = '/opt/container-explorer/bin/ce'
+CE_SUPPORT_FILE = '/opt/container-explorer/etc/supportcontainer.yaml'
+
+
+class ContainerdEnumerationTask(TurbiniaTask):
+  """Enumerate containerd containers on Linux."""
+
+  REQUIRED_STATES = [state.ATTACHED, state.MOUNTED]
+
+  def list_containers(self, evidence, _, detailed_output=False):
+    """List containerd containers in the evidence.
+
+    Args:
+      evidence (Evidence): Input evidence to be processed.
+      result (TurbiniaTaskResult): Object to store logs.
+
+    Returns:
+      list(dict): Containers information
+    """
+    config.LoadConfig()
+    containers = None
+    image_path = evidence.mount_path
+
+    outputfile = tempfile.mkstemp()[1]
+    list_cmd = [
+        'sudo', CE_BINARY, '--support-container-data', CE_SUPPORT_FILE,
+        '--output', 'json', '--output-file', outputfile, '--image-root',
+        image_path, 'list', 'containers'
+    ]
+    log.info(f'Running {list_cmd}')
+
+    try:
+      #json_data = subprocess.check_output(list_cmd).decode('utf-8')
+      subprocess.check_call(list_cmd)
+      with open(outputfile, 'r') as fh:
+        json_data = fh.read()
+        if json_data:
+          containers = json.loads(json_data)
+    except json.JSONDecodeError as e:
+      raise TurbiniaException(
+          f'Error decoding container-explorer output: {e}') from e
+    except subprocess.CalledProcessError as e:
+      raise TurbiniaException(f'container-explorer return error: {e}') from e
+
+    return self._list_containers_result(containers, detailed_output)
+
+  def _list_containers_result(self, containers, detailed_output):
+    """ Determine and return containers information.
+
+    Args:
+      containers (list(dict)): Containers information
+      detailed_output (bool): Check if detailed output is required.
+
+    Returns:
+      list(dict): List containers basic or complete output.
+    """
+    if not containers:
+      return containers
+
+    if detailed_output:
+      return containers
+
+    basic_fields = [
+        'Namespace', 'Image', 'ContainerType', 'ID', 'Hostname', 'CreatedAt'
+        'Labels'
+    ]
+    basic_containers = []
+
+    for container in containers:
+      basic_container = {}
+      for key, value in container.items():
+        if key not in basic_fields:
+          continue
+        basic_container[key] = value
+      basic_containers.append(basic_container)
+
+    return basic_containers
+
+  def run(self, evidence, result):
+    """Run ContainerdEnumerationTask.
+
+    Args:
+      evidence (Evidence): Evidence to process.
+      result (TurbiniaTaskResult): The object to place task result.
+
+    Returns:
+      TurbiniaTaskResult object
+    """
+    summary = ''
+    success = False
+
+    image_path = evidence.mount_path
+    if not image_path:
+      raise TurbiniaException('Evidence is not mounted')
```

Comment:
Can you add the evidence name and the path to the error text as well?

Also, instead of raising an exception here, we should close out the result (e.g. with `result.close()`) and set the the right success/status values (and return the result).

Reply:
Replaced exception with result.close. The status message contains evidence name and source_path.